lesson18/src/js/main.js

```diff
@@ -0,0 +1,275 @@
+  pagenationBullet.setAttribute('aria-current', false);
+  const pageNations = document.querySelector(`[data-pagenation-index="${target}"]`);
+  pageNations.setAttribute("aria-current", true);
+
+}
+
+function clickPagenationBullet(){
+  const pagenationBullet = [...document.getElementsByClassName("bar")];
+  pagenationBullet.forEach((targetButton) => {
+    targetButton.addEventListener('click',(e) => {
+      const activePagenationBullet = document.querySelector('[aria-current="true"]');
+      const clickedPagenationBulletNo = e.currentTarget.getAttribute('data-pagenation-index');
+      activePagenationBullet.setAttribute('aria-current','false');
+      e.currentTarget.setAttribute('aria-current','true');
+      switchSlider(clickedPagenationBulletNo);
+      switchPagination(clickedPagenationBulletNo);
+      resetAutoPlaySlide();
+    })
+  })
+
+}
+
+function toggleButtonDisabled(index) {
+  const slideImages = [...document.getElementsByClassName("slider__item")];
+  const firstIndex = 1;
+  const lastIndex = slideImages.length;
+  const prevButton = document.getElementById("js-button_previous");
+  const nextButton = document.getElementById("js-button_next");
+
+  prevButton.disabled = Number(index) === firstIndex;
+  nextButton.disabled = Number(index) === lastIndex;
+}
+
+function renderSlidContents(slideImageData){
+  renderSlideImage(slideImageData);
+  renderPagenation(slideImageData.length);
+  renderPagenationBullet(slideImageData.length);
+  renderSlideButton();
+}
+
+function addSlideAction(){
+  addSwitchButtonEvent();
+  clickPagenationBullet();
+  autoPlayslide();
+}
+
+function callImageData() {
+  return new Promise((resolve) => {
+    setTimeout(() => {
+      resolve(
+        fetchImageData(
+//https://api.json-generator.com/templates/IZjWl012CAMD/data?access_token=b0154huvd1stffra1six9olbgg34r4zofcqgwzfl : image = 0
+          //https://api.json-generator.com/templates/9tm12BO1y5Xx/data?access_token=b0154huvd1stffra1six9olbgg34r4zofcqgwzfl&status=503 : 503
+          "https://api.json-generator.com/templates/9tm12BO1y5Xx/data?access_token=b0154huvd1stffra1six9olbgg34r4zofcqgwzfl"
+        )
+      );
+    }, 3000);
+  });
+}
+
+async function fetchImageData(URL) {
+  const response = await fetch(URL);
+  if (response.ok) {
+    const json = await response.json();
+    return json;
+  } else {
+    console.error(`${response.status}:${response.statusText}`);
+    displayErrorMassage("サーバの通信に失敗しました");
+  }
+}
+
+async function getSlideImage(){
+  showLoadingImage();
+  renderSlideArea();
+  try{
+     const json = await callImageData();
+     const slideData = json.slide;
+    if(slideData.length > 0){
+      renderSlidContents(slideData);
+      addSlideAction()
+    }else{
+      addNoimage();
+    }
+  }catch{
+    console.error('表示する画像がありませんでした');
+  }finally{
+    removeLoading();
+  }
+}
+
+async function init() {
+  const slideContents = await getSlideImage();
+  slideContents && autoPlayslide();
+}
+
+init();
```

Comment:
if this function is main function, don't write some function this line below.

Reply:
I moved the description location to the end.
By having the main init () function in the middle,
I also understand that it can cause problems.

[b083bf5](https://github.com/nyattsu72/hanson-frontend-cource/pull/26/commits/b083bf562404613659f7c0f8e46a619d76f3e599)